ledger/participant-integration-api/src/main/scala/platform/store/backend/h2/H2Table.scala

```diff
@@ -33,11 +36,20 @@ private[h2] object H2Table {
                 }
                 preparedStatement.addBatch()
               }
-              preparedStatement.executeBatch()
+              retry(10) {
+                preparedStatement.executeBatch()
+              }
               preparedStatement.close()
             }
     }
 
+  @tailrec
+  private def retry[T](maxRetry: Int)(body: => T): T = Try(body) match {
+    case Success(result) => result
+    case Failure(t) if maxRetry <= 0 => throw t
+    case Failure(_) => retry(maxRetry - 1)(body)
+  }
```

Comment:
Sorry for maybe out of context question, but would backoff retry strategy be better here?

Reply:
Thank you for the explanation @nmarton-da 👍